testing/integration/beat_receivers_test.go

```diff
@@ -225,198 +238,68 @@ func TestAgentMonitoring(t *testing.T) {
 		t.Cleanup(func() {
 			if t.Failed() {
-				t.Logf("Contents of agent config file:\n%s\n", string(configOTelContents))
+				t.Logf("policy was %s", string(updatedPolicyBytes))
 			}
 		})
-		err = fixture.Configure(ctx, configOTelContents)
+
+		// 3. Install without enrolling in fleet
+		fixture, err := define.NewFixtureFromLocalBuild(t, define.Version())
 		require.NoError(t, err)
 
-		// Get the timestamp before restarting. Required to separate logs from agent and otel
-		timestamp := time.Now().UTC().Format("2006-01-02T15:04:05.000Z")
+		err = fixture.Prepare(ctx)
+		require.NoError(t, err, "error preparing fixture")
 
-		fmt.Println(time.Now())
-		// Restart elastic-agent
-		output, err = fixture.Exec(ctx, []string{"restart"})
-		require.NoErrorf(t, err, "error restarting agent: %s\ncombinedoutput:\n%s", err, string(output))
+		err = fixture.Configure(ctx, updatedPolicyBytes)
+		require.NoError(t, err, "error configuring fixture")
 
-		require.Eventually(t, func() bool {
-			err = fixture.IsHealthy(ctx)
-			if err != nil {
-				t.Logf("waiting for agent healthy: %s", err.Error())
-				return false
-			}
-			return true
-		}, 30*time.Second, 1*time.Second)
+		output, err := fixture.InstallWithoutEnroll(ctx, &installOpts)
+		require.NoErrorf(t, err, "error install withouth enroll: %s\ncombinedoutput:\n%s", err, string(output))
+
+		require.EventuallyWithT(t, func(collect *assert.CollectT) {
+			status, statusErr := fixture.ExecStatus(ctx)
+			assert.NoError(collect, statusErr)
+			// agent should be healthy
+			assert.Equal(collect, int(cproto.State_HEALTHY), status.State)
+			// we should have no normal components running
+			assert.Zero(collect, len(status.Components))
+
+			// we should have filebeatreceiver and metricbeatreceiver running
+			otelCollectorStatus := status.Collector
+			assert.Equal(collect, int(cproto.CollectorComponentStatus_StatusOK), otelCollectorStatus.Status)
+			pipelineStatusMap := otelCollectorStatus.ComponentStatusMap
+
+			// we should have 3 pipelines running: filestream for logs, http metrics and beats metrics
+			assert.Equal(collect, 3, len(pipelineStatusMap))
+
+			fileStreamPipeline := "pipeline:logs/_agent-component/filestream-monitoring"
+			httpMetricsPipeline := "pipeline:logs/_agent-component/http/metrics-monitoring"
+			beatsMetricsPipeline := "pipeline:logs/_agent-component/beat/metrics-monitoring"
+			assert.Contains(collect, pipelineStatusMap, fileStreamPipeline)
+			assert.Contains(collect, pipelineStatusMap, httpMetricsPipeline)
+			assert.Contains(collect, pipelineStatusMap, beatsMetricsPipeline)
+
+			// and they should be healthy
+			assert.Equal(collect, int(cproto.CollectorComponentStatus_StatusOK), pipelineStatusMap[fileStreamPipeline].Status)
+			assert.Equal(collect, int(cproto.CollectorComponentStatus_StatusOK), pipelineStatusMap[httpMetricsPipeline].Status)
+			assert.Equal(collect, int(cproto.CollectorComponentStatus_StatusOK), pipelineStatusMap[beatsMetricsPipeline].Status)
```

Comment:
Once we've made it to this point, can we also make sure that none of the sub-process monitoring components are there as well?

Reply:
Probably easier to just check the whole map recursively.